aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,402 @@
+        is_incoming = default_is_incoming if message[2] is None else message[2]
+
+        default_target = message_id - 1
+        target = default_target if message[3] is None else message[3]
+        return is_incoming, target
+
+    @staticmethod
+    def _non_initial_incoming_message_dialogue_reference(
+        dialogue: Dialogue,
+    ) -> Tuple[str, str]:
+        """
+        Specifies the dialogue reference of a non-initial incoming message for a dialogue.
+
+        It uses a complete version of the reference in the dialogue if it is incomplete,
+        otherwise it uses the reference in the dialogue.
+
+        :param dialogue: the dialogue to which the incoming message is intended
+        :return: its dialogue reference
+        """
+        dialogue_reference = (
+            dialogue.dialogue_label.dialogue_reference[0],
+            Dialogues._generate_dialogue_nonce()  # pylint: disable=protected-access
+            if dialogue.dialogue_label.dialogue_reference[1]
+            == Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+            else dialogue.dialogue_label.dialogue_reference[1],
+        )
+        return dialogue_reference
+
+    def _extract_message_fields(
+        self,
+        message: DialogueMessage,
+        messages: Tuple[DialogueMessage, ...],
+        last_is_incoming: bool,
+    ):
+        performative = message[0]
+        contents = message[1]
+        message_id = messages.index(message) + 1
+        is_incoming, target = self._provide_unspecified_fields(
+            message, last_is_incoming=last_is_incoming, message_id=message_id,
+        )
+        return performative, contents, message_id, is_incoming, target
+
+    def prepare_skill_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[DialogueMessage, ...],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of DialogueMessages.
+        For every DialogueMessage (performative, contents, is_incoming, target):
+            - if 'is_incoming' is not provided: for the first message it is assumed False (outgoing),
+            for any other message, it is the opposite of the one preceding it.
+            - if 'target' is not provided: for the first message it is assumed 0,
+            for any other message, it is the index of the message before it in the tuple of messages + 1.
+
+        :param dialogues: a dialogues class
+        :param counterparty: the message_id
+        :param messages: the dialogue_reference
+
+        :return: the created incoming message
+        """
+        if not len(messages) >= 1:
+            raise AEAEnforceError("the list of messages must be positive.")
+
+        (
+            performative,
+            contents,
+            message_id,
+            is_incoming,
+            target,
+        ) = self._extract_message_fields(messages[0], messages, True)
```

Comment:
```suggestion
        ) = self._extract_message_fields(messages[0], index=0, last_is_incoming=True)
```

Reply:
(why is last_is_incoming=True by default, should be None?)

Reply:
The default `is_incoming` value, if not specified, is the opposite of the `is_incoming` of the last message, and for the first message it is outgoing. 

For the first message, the `last_is_incoming` above is set to `True` to make its `is_incoming` (if unspecified) False.

Reply:
Done